src/network/networkd-ndisc.c

```diff
@@ -1192,12 +1208,26 @@ static int ndisc_router_handler(Link *link, sd_ndisc_router *rt) {
                 log_link_debug(link, "Received RA without timestamp, ignoring.");
                 return 0;
         }
+
+        r = ndisc_drop_outdated(link, timestamp_usec, NULL);
         if (r < 0)
                 return r;
 
-        r = ndisc_drop_outdated(link, timestamp_usec);
+        r = sd_ndisc_router_get_lifetime(rt, &router_lifetime_sec);
         if (r < 0)
-                return r;
+                return log_link_warning_errno(link, r, "Failed to get lifetime of RA message: %m");
+
+         /* https://datatracker.ietf.org/doc/html/rfc4861
+          * Router Lifetime: A Lifetime of 0 indicates that the router is not a default router
+          * and SHOULD NOT appear on the default router list.
+          */
+        if (router_lifetime_sec == 0) {
+                log_link_debug(link, "Received RA with lifetime = 0, dropping configurations.");
+
+                r = ndisc_drop_outdated(link, USEC_INFINITY, &router);
+                if (r < 0)
+                        log_link_warning_errno(link, r, "Failed to process RA with zero lifetime, ignoring:: %m");
```

Comment:
Doubled `:` in the log message.

Also, `return 0;` in this `if` block is mistakenly dropped:
```c
if (router_lifetime_sec == 0) {
        log_link_debug(...);

        r = ndisc_drop_outdated(...);
        if (r < 0)
                log_link_warning_errno(...);

        return 0;
}
```

Reply:
Opps sry missed it